### DISCUSSION TITLE
Config file creation via Custom Action Best Practices

### DISCUSSION THREAD

**Discussion by thebruuu** [wixtoolset/issues#8777] [Questions]
Hello 

I need the installer to write a config file to store on an hard coded directory, which will contain the installation directory.
What's the best approach?
Writing a powershell script and run it?
Writing a VB/Java script and run it?
Writing a DLL, include it in the project, run it?

Thank You

Working with v5

**Comment by bevanweiss** [comment]
It depends how this configuration file will be used.
If it's just defaults, and you might want to update it with later versions / patches of your application, then just having it as another file deployed using the InstallFile standard action would make sense.  In such a situation you'd have another file generated by the app that would be the user changes to the defaults.

If the user needs to modify the original file themselves, then it's normally best to keep it away from the MSI installer entirely.
Perhaps the application itself could create the default version on first run?  Or you could mark a dedicated component for the file as Permanent.

https://stackoverflow.com/questions/17759230/marking-config-file-as-permanent
